packages/salesforcedx-visualforce-markup-language-server/test/completion.test.ts

```diff
@@ -564,4 +569,118 @@ describe('HTML Completion', () => {
     testTagCompletion('<div><br><span></span></|', 'div>');
     testTagCompletion('<div><h1><br><span></span><img></| </h1></div>', 'h1>');
   });
+
+  // Visualforce
+  //////////////
+
+  function getCompletionSuggestions(value: string): CompletionList {
+    const offset = value.indexOf('|');
+    value = value.substr(0, offset) + value.substr(offset + 1);
+
+    const ls = htmlLanguageService.getLanguageService();
+
+    const document = TextDocument.create(
+      'test://test/test.page',
+      'visualforce',
+      0,
+      value
+    );
+    const position = document.positionAt(offset);
+    const vfDoc = ls.parseHTMLDocument(document);
+    const list = ls.doComplete(document, position, vfDoc);
+    return list;
+  }
+
+  describe('Visualforce Completions', () => {
+    // This list is from https://developer.salesforce.com/docs/atlas.en-us.pages.meta/pages/pages_compref_map.htm
+    it('Should display basic tags', () => {
+      const expectedNamespaces = [
+        'analytics',
+        'apex',
+        'chatter',
```

Comment:
One useful note: chatteranswer is not here. I think it's OK to remove it because it will be retired in Spring '18.

See https://help.salesforce.com/articleView?id=000240491&type=1. Useful to know this when doing QA.